python/cuml/ensemble/randomforestregressor.pyx

```diff
@@ -378,54 +379,47 @@ class RandomForestRegressor(Base):
                              " please read the documentation")
 
     def _obtain_treelite_handle(self):
-        cdef ModelHandle cuml_model_ptr = NULL
+        cdef uintptr_t tl_handle_int
+        cdef ModelHandle tl_handle = NULL
         cdef RandomForestMetaData[float, float] *rf_forest = \
             <RandomForestMetaData[float, float]*><uintptr_t> self.rf_forest
-        cdef unsigned char[::1] model_pbuf_mv = self.model_pbuf_bytes
-        cdef vector[unsigned char] model_pbuf_vec
-        with cython.boundscheck(False):
-            model_pbuf_vec.assign(& model_pbuf_mv[0],
-                                  & model_pbuf_mv[model_pbuf_mv.shape[0]])
         if self.treelite_handle is None:
-            task_category = CLASSIFICATION_MODEL
-            build_treelite_forest(
-                & cuml_model_ptr,
-                rf_forest,
-                <int> self.n_cols,
-                <int> task_category,
-                model_pbuf_vec)
-            mod_ptr = <uintptr_t> cuml_model_ptr
-            self.treelite_handle = ctypes.c_void_p(mod_ptr).value
+            if self.model_bytes:  # bytes -> Treelite
+                tl_handle_int = treelite_deserialize(self.model_bytes['header'],
+                                                     self.model_bytes['frames'])
+                tl_handle = <ModelHandle>tl_handle_int
+            else:                 # RF -> Treelite
+                task_category = CLASSIFICATION_MODEL
+                build_treelite_forest(
+                    &tl_handle,
+                    rf_forest,
+                    <int> self.n_cols,
+                    <int> task_category)
+            self.treelite_handle = ctypes.c_void_p(<uintptr_t>tl_handle).value
         return self.treelite_handle
 
-    def _get_protobuf_bytes(self):
+    def _get_model_bytes(self):
         """
-        Returns the self.model_pbuf_bytes.
+        Returns the self.model_bytes.
```

Comment:
Maybe clarify the type and update the rest of docstring: "Returns the treelite binary format representation of this model."